ibis/backends/flink/tests/conftest.py

```diff
@@ -135,3 +180,14 @@ def temp_view(con) -> str:
     yield name
 
     con.drop_view(name, force=True)
+
+
+@pytest.fixture(autouse=True)
+def reset_con(con):
+    yield
+    tables_to_drop = list(set(con.list_tables()) - set(TEST_TABLES.keys()))
+    for table in tables_to_drop:
+        con.drop_table(table, force=True)
+    views_to_drop = list(set(con.list_views()) - set(TEST_TABLES.keys()))
+    for view in views_to_drop:
+        con.drop_view(view, temp=True, force=True)
```

Comment:
Hm, what is this `temp` argument to `drop_view`? It's a bit odd to have to specify the lifetime of a view when dropping it. Is that a flink idiosyncracy?

Reply:
When tables are created from in-memory objects in Flink (e.g. from a pandas dataframe), a temporary view is created. Therefore, when resetting the connection, we also need to explicitly drop the temporary view that was created as well

Reply:
Right, what I am wondering is why `con.drop_view(view, force=True)` doesn't suffice for that. Why is the additional `temp=True` needed?

Reply:
Sorry I misunderstood your question. If you do `con.drop_view(view)` you would get the following error: 
```
org.apache.flink.table.api.ValidationException: Temporary view with identifier '`default_catalog`.`default_database`.`temp`' exists. Drop it first before removing the permanent view.
```

```python
> import pandas as pd
> import ibis
> from pyflink.table import EnvironmentSettings, TableEnvironment
> from pyflink.common import Configuration

> env_settings = EnvironmentSettings.in_streaming_mode()
> table_env = TableEnvironment.create(env_settings)

> table_config = table_env.get_config()
> config = Configuration()
> config.set_string("parallelism.default", "1")
> table_config.add_configuration(config)

> con = ibis.flink.connect(table_env)

> df = pd.DataFrame(
        {
            "row_time": [
                pd.to_datetime("2020-04-15 12:02"),
                pd.to_datetime("2020-04-15 12:06"),
                pd.to_datetime("2020-04-15 12:03"),
            ],
            "num": [1, 2, 3],
            "id": ["L1", "L2", "L3"],
        }
    )
> table = con.create_table("temp", df, temp=True)
> print(f"Tables: {con.list_tables(temp=True)}")
> print(f"Views: {con.list_views()}")
> print(f"Temp views: {con.list_views(temp=True)}")
Tables: ['temp']
Views: ['temp']
Temp views: ['temp']

> con.drop_view("temp")
Py4JJavaError: An error occurred while calling o8.executeSql.
: org.apache.flink.table.api.ValidationException: Temporary view with identifier '`default_catalog`.`default_database`.`temp`' exists. Drop it first before removing the permanent view.
	at org.apache.flink.table.catalog.CatalogManager.dropTableInternal(CatalogManager.java:1231)
	at org.apache.flink.table.catalog.CatalogManager.dropView(CatalogManager.java:1218)
	at org.apache.flink.table.operations.ddl.DropViewOperation.execute(DropViewOperation.java:73)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeInternal(TableEnvironmentImpl.java:1092)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeSql(TableEnvironmentImpl.java:735)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.flink.api.python.shaded.py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at org.apache.flink.api.python.shaded.py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
	at org.apache.flink.api.python.shaded.py4j.Gateway.invoke(Gateway.java:282)
	at org.apache.flink.api.python.shaded.py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at org.apache.flink.api.python.shaded.py4j.commands.CallCommand.execute(CallCommand.java:79)
	at org.apache.flink.api.python.shaded.py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.base/java.lang.Thread.run(Thread.java:829)

> con.drop_view("temp", force=True)
Py4JJavaError: An error occurred while calling o8.executeSql.
: org.apache.flink.table.api.ValidationException: Temporary view with identifier '`default_catalog`.`default_database`.`temp`' exists. Drop it first before removing the permanent view.
	at org.apache.flink.table.catalog.CatalogManager.dropTableInternal(CatalogManager.java:1231)
	at org.apache.flink.table.catalog.CatalogManager.dropView(CatalogManager.java:1218)
	at org.apache.flink.table.operations.ddl.DropViewOperation.execute(DropViewOperation.java:73)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeInternal(TableEnvironmentImpl.java:1092)
	at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeSql(TableEnvironmentImpl.java:735)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.flink.api.python.shaded.py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at org.apache.flink.api.python.shaded.py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
	at org.apache.flink.api.python.shaded.py4j.Gateway.invoke(Gateway.java:282)
	at org.apache.flink.api.python.shaded.py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at org.apache.flink.api.python.shaded.py4j.commands.CallCommand.execute(CallCommand.java:79)
	at org.apache.flink.api.python.shaded.py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.base/java.lang.Thread.run(Thread.java:829)

> con.drop_view("temp", temp=True)

> con.list_views()
[]

> con.list_views(temp=True)
[]
```